plutus-ledger-api/src/PlutusLedgerApi/Common/SerialisedScript.hs

```diff
@@ -155,6 +159,17 @@ serialiseUPLC =
   -- need to be careful about introducing a working version
   toShort . BSL.toStrict . serialise . SerialiseViaFlat . UPLC.UnrestrictedProgram
 
+{- | Hash a 'SerialisedScript' with the given version prefix. Each prefix corresponds
+to a specific version of the Plutus language.
+
+As of PlutusV3, the Plutus language versions and corresponding prefixes are as follows:
+  PlutusV1 -> 0x1
+  PlutusV2 -> 0x2
+  PlutusV3 -> 0x3
+-}
+hashScriptWithPrefix :: Word8 -> SerialisedScript -> BS.ByteString
+hashScriptWithPrefix prefix script = Hash.blake2b_224 (BS.singleton prefix <> fromShort script)
```

Comment:
Is the RHS the same as `Hash.blake2b_224 (BS.cons prefix (fromShort script))`?

Reply:
Yes, changed it as suggested.